Editor plugin gets dynamic CONFIG

jon-heard opened the pull request:
First, I apologize for bringing an update to this editor plugin system so soon after its launch.

### The problem:
When submitting to bipsi-hacks, Sean pointed out that changes to an editor plugin's settings aren't applied until the page is refreshed.  This is because CONFIG is created at plugin setup, and does NOT update to match settings changes.  This makes sense for playback, where the user doesn't change settings.  For edit-time, though, special consideration is needed to use the latest settings without a page refresh.

### The solution:
After consideration, I came up with the solution in this pull request.  This sets up CONFIG quite differently for editor code than for playback code.  For playback, CONFIG is the same as it always was.  For editor-time, CONFIG is an empty object with its prototype set to the plugin's _actual_ event.  This way it can always provide the latest settings, and its prototype can be re-set when the event object is replaced (for undo/redo).

### An alternative
If this solution isn't viable for some reason I have a minimal code-change alternate: simply give the plugin it's event id.  This isn't very nice, though.  The editor code must call `findEventById()` to grab the latest settings (a relatively heavy task).  Meanwhile, the playback code still uses CONFIG, meaning different code sections get the settings in different ways.

jon-heard commented:
I'm attaching the latest version of the "fullscreen" plugin.  It is designed to work with this pull request.  To try it, use the fullscreen hotkey as normal and note that it works, then change the plugin's "fullscreen-hotkey" field to something like "Home" or "Backspace" and try it again, WITHOUT refreshing the webpage.  The new hotkey should effect take immediately.

[editor fullscreen.txt](https://github.com/Ragzouken/bipsi/files/12094616/editor.fullscreen.txt)

Ragzouken commented on src/scripts/maker.js:
this class shouldn't know about the editor or plugins, so you should be able to replace these calls with on in the editor's listener for `this.stateManager`'s `change` event (somewhere around line ~1530). that's where everything gets updated when there's a state change

jon-heard replied:
I've made this requested change

Ragzouken reviewed (changes requested):
this seems acceptable, aside from the change i requested

jon-heard commented:
I've made the requested change.  In addition:

- There is an edge case where the user either repeatedly hits undo to before loading the event or deletes the event, then hits undo or redo, where the event is still deleted.  I've adjusted how this is handled.  Previously, the config settings are cleared to defaults in this scenario.  Now, the config settings remain at what they were when the event was last available.
- I've moved the config-handling logic into BipsiEditor, since it's all dealing with BipsiEditor.editorPluginConfigs.

Sorry if these changes require a lot of extra effort on your part.
